index.js

```diff
@@ -1,88 +1,117 @@
 // ***** Question 1 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 1", "color: red") 
+let counter = 1
+counter = 2
+console.log("%cQuestion 1", "color: red") 
 
-// console.log(counter) 
+console.log(counter) 
 // // => 2
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 // ***** Question 2 *****
 // *** Uncomment the lines below to test
 // *** after testing, comment the line causing the error back in 
 // *** otherwise, the error will stop the rest of your code from running
-// console.log("%cQuestion 2", "color: red") 
-
-// console.log(name)
-// => "Raffy"
+console.log("%cQuestion 2", "color: red") 
+const name = "Eugene"
+console.log(name)
+// // => "Raffy"
 // name = "Not Raffy" 
-// => TypeError
-// console.log("%c----------", "color: red") 
+// // => TypeError
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 3 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 3", "color: red") 
-
-// console.log(drinkWater(12))
+console.log("%cQuestion 3", "color: red") 
+function drinkWater(currentThirstLevel){
+  console.log("Man I sure am thirsty")
+  currentThirstLevel -= 1
+  console.log("Ahh that hits the spot")
+  return currentThirstLevel
+}
+// def drink_water(current_thirst_level)
+//   puts "Man I sure am thirsty"
+//   current_thirst_level -= 1
+//   puts "Ahh that hits the spot"
+//   current_thirst_level
+// end
+console.log(drinkWater(12))
 // // => "Man I sure am thirsty"
 // // => "Ahh that hits the spot"
 // // => 11
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
-
-// console.log(sameSameButDifferent(5, 5)) 
+console.log("%cQuestion 4", "color: red") 
+function sameSameButDifferent (num, maybeNum){
+  if (num === maybeNum){
+    return "same same"
+  } else if (num == maybeNum){
+    return "same same (but different)"
+  } else {
+    return "different"
+  }
+}
+console.log(sameSameButDifferent(5, 5)) 
```

Comment:
Nice! Another solution would be to convert the string to a number with parseInt to avoid `== `:

`} else if (num === parseInt(maybeNum)) {`